text/0004-new-pre-coordicide-consensus/0004-new-pre-coordicide-consensus.tex.md

```diff
@@ -0,0 +1,191 @@
+ 
+1. $lsmi-ytrsi(v)>C_1$    
+2. $lsmi-otrsi(v)>M$    
+3. both $v_i$ satisfy $lsmi-otrsi(v_i)>C_2$
+4. at least one $v_i$ has a score of 0 (to enforce monotonicity)
+
+Else Score 1 will be given if exactly one $v_i$ satisfies $lsmi -otrsi(v_i) \leq C_2$
+
+Else Score 2 will given.
+
+#### Performing the weighted random selection
+A node should have in memory the entire set of `tips` and their scores:
+
+1. Sum up the total score of the tips.
+2. Pick a random number, `r` between 1 and the sum.
+3. Traverse through the set of tips and subtract the score of each tip from `r`. Stop when you reach 0 or lower.
+
+## Compass Tip Selection
+TBD
+
+
+# Drawbacks
+
+## White Flag
+1. If we ever want to supply a proof that a `value transfer` is valid and approved we can't do so by merely supplying the path from the bundle to the approving milestone as before. A proof will require to have all the transactions that are in the past cone of the approving milestone to the genesis. However, up until now we never required to give such proofs. If we want to do easy proofs we can create merkle trees from apporved bundle hashes and add them to milestones.
+2. Everything that is `seen` is part of the tangle, including double-spend attempts. Meaning we define that possibly malicious data will be saved as part of the consensus set of the tangle.
+
+## Node Tip Selection
+1. Timestamp proposal relies on local values that differ from node to node. This can theoretically cause some tips to appear lazy to certain nodes in the network, but not lazy to others. This may lead to create competing subtangles, thus the the milestone proposal is superior. However node implementation that would like to later switch to coordicide may still consider implementing it.
+
+2. With both proposals users may have to reattach if they create lazy tips. This is not so bad, becuase honest users should never create lazy tips unless they encounter access barriers to the network, i.e. bad connection or congested network.
+
+## Compass Tip Selection
+TBD
+
+# Rationale and alternatives
+
+- The previous design tried to stay as close as it can to the design outlayed by the [original iota whitepaper](https://assets.ctfassets.net/r1dr6vzfxhev/2t4uxvsIqk0EUau6g2sw0g/45eae33637ca92f85dd9f4a3a218e1ec/iota1_4_3.pdf). Due to the pivot to [coordicide](https://files.iota.org/papers/Coordicide_WP.pdf) it was decided to find a design that can maximize the benefits of the current coordinator based mainnet in the meanwhile. The design proposed here embraces the power of the coordinator rather than utilize it as an auxilary tool intended to be removed. This will give us all the beneifits described in [motivation](#motivation).
+
+# Unresolved questions
+
+## Whiteflag
+Since nodes will try to sort out conflcits themselves perhaps it will be wise to add more prtoection against forks. In a seperate RFC we can maybe define additional data that can be added to milestones to prevent that.
```

Comment:
```suggestion
Since nodes will try to sort out conflicts themselves perhaps it will be wise to add more protection against forks. In a separate RFC we can maybe define additional data that can be added to milestones to prevent that.
```